src/Shared/EncodingUtilities.cs

```diff
@@ -14,8 +14,15 @@ namespace Microsoft.Build.Shared
     /// </summary>
     internal static class EncodingUtilities
     {
+        internal static readonly Encoding Utf8WithoutBom = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+
         private static Encoding s_currentOemEncoding;
 
+        internal const string UseUtf8Always = "ALWAYS";
+        internal const string UseUtf8Never = "NEVER";
+        internal const string UseUtf8Detect = "DETECT";
+        internal const string UseUtf8System = "SYSTEM";
```

Comment:
Those would be nicer as an enum.

Reply:
Mmmmmm, enums 🍰

Reply:
This is moved code, too. I'm not sure why it wasn't an enum in the first place but I'd rather leave it as-is now.